src/vivarium/framework/values.py

```diff
@@ -298,12 +381,12 @@ def on_post_setup(self, _):
     def register_value_producer(
         self,
         value_name: str,
-        source: Callable,
-        requires_columns: List[str] = (),
-        requires_values: List[str] = (),
-        requires_streams: List[str] = (),
-        preferred_combiner: Callable = replace_combiner,
-        preferred_post_processor: Callable = None,
+        source: ValueSource,
+        requires_columns: Iterable[str] = (),
```

Comment:
Is it the case that for all of these all we do is iterate over them and that the order doesn't matter?

Reply:
I don't think it does--_convert_dependencies only loops over them, and I don't think the order should matter. I didn't look over Resource super closely though. If you think its safer I could type them as Sequence instead?